android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/EventHandler.kt

```diff
@@ -0,0 +1,67 @@
+package com.devcycle.sdk.android.eventsource
+
+import timber.log.Timber
+import com.devcycle.sdk.android.eventsource.EventHandler
+
+class Handler: EventHandler {
+    fun init() {}
+
+    /**
+     * EventSource calls this method when the stream connection has been opened.
+     * @throws Exception throwing an exception here will cause it to be logged and also sent to [.onError]
+     */
+    override fun onOpen() {
+        Timber.d("OPEN")
+    }
+
+    /**
+     * EventSource calls this method when the stream connection has been closed.
+     *
+     *
+     * This method is *not* called if the connection was closed due to a [ConnectionErrorHandler]
+     * returning [ConnectionErrorHandler.Action.SHUTDOWN]; EventSource assumes that if you registered
+     * such a handler and made it return that value, then you already know that the connection is being closed.
+     *
+     *
+     * There is a known issue where `onClosed()` may or may not be called if the stream has been
+     * permanently closed by calling `close()`.
+     *
+     * @throws Exception throwing an exception here will cause it to be logged and also sent to [.onError]
+     */
+    override fun onClosed() {
+        Timber.d("CLOSED")
+    }
+
+    /**
+     * EventSource calls this method when it has received a new event from the stream.
+     * @param event the event name, from the `event:` line in the stream
+     * @param messageEvent a [MessageEvent] object containing all the other event properties
+     * @throws Exception throwing an exception here will cause it to be logged and also sent to [.onError]
+     */
+    @Throws(Exception::class)
+    override fun onMessage(event: String?, messageEvent: MessageEvent?) {
```

Comment:
how does the client react to new messages received here? you should wire that up here unless someone is doing it in a later PR

Reply:
That would be done in the ticket thats for refetching the config when a new message is received